data-prepper-plugin-framework/src/main/java/org/opensearch/dataprepper/plugin/PluginConfigurationErrorHandler.java

```diff
@@ -0,0 +1,91 @@
+package org.opensearch.dataprepper.plugin;
+
+import com.fasterxml.jackson.databind.JsonMappingException;
+import com.fasterxml.jackson.databind.exc.UnrecognizedPropertyException;
+import org.apache.commons.text.similarity.LevenshteinDistance;
+import org.opensearch.dataprepper.model.configuration.PluginSetting;
+import org.opensearch.dataprepper.model.plugin.InvalidPluginConfigurationException;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Named
+public class PluginConfigurationErrorHandler {
+
+    static final String UNRECOGNIZED_PROPERTY_EXCEPTION_FORMAT = "Parameter \"%s\" for plugin \"%s\" does not exist. Available options include %s.";
+    static final String JSON_MAPPING_EXCEPTION_FORMAT = "Parameter \"%s\" for plugin \"%s\" is invalid: %s";
+    static final String GENERIC_PLUGIN_EXCEPTION_FORMAT = "Plugin \"%s\" is invalid: %s";
+
+    static final Integer MIN_DISTANCE_TO_RECOMMEND_PROPERTY = 3;
+
+    private final LevenshteinDistance levenshteinDistance;
+
+    @Inject
+    public PluginConfigurationErrorHandler(final LevenshteinDistance levenshteinDistance) {
+        this.levenshteinDistance = levenshteinDistance;
+    }
+
+    public void handleException(final PluginSetting pluginSetting, final Exception e) {
+        if (e.getCause() instanceof UnrecognizedPropertyException) {
+            handleUnrecognizedPropertyException((UnrecognizedPropertyException) e.getCause(), pluginSetting);
+        } else if (e.getCause() instanceof JsonMappingException) {
+            handleJsonMappingException((JsonMappingException) e.getCause(), pluginSetting);
+        }
+
+        throw new InvalidPluginConfigurationException(
+                String.format(GENERIC_PLUGIN_EXCEPTION_FORMAT, pluginSetting.getName(), e.getMessage()));
+    }
+
+    private void handleJsonMappingException(final JsonMappingException e, final PluginSetting pluginSetting) {
+        final String parameterPath = getParameterPath(e.getPath());
+
+        final String errorMessage = String.format(JSON_MAPPING_EXCEPTION_FORMAT,
+                parameterPath, pluginSetting.getName(), e.getOriginalMessage());
+
+        throw new InvalidPluginConfigurationException(errorMessage);
+    }
+
+    private void handleUnrecognizedPropertyException(final UnrecognizedPropertyException e, final PluginSetting pluginSetting) {
+        String errorMessage = String.format(UNRECOGNIZED_PROPERTY_EXCEPTION_FORMAT,
+                getParameterPath(e.getPath()),
+                pluginSetting.getName(),
+                e.getKnownPropertyIds());
+
+        final Optional<String> closestRecommendation = getClosestField(e);
+
+        if (closestRecommendation.isPresent()) {
+            errorMessage += " Did you mean \"" + closestRecommendation.get() + "\"?";
```

Comment:
I actually feel like it is easier to see the recommendation when it's at the end with just a glance. If the entire message is read in order then the recommendation coming first is good, but it does seem easier to see when it's at the end.

Reply:
```
1. entry-pipeline:processor:parse_json: caused by: Parameter "destintio" for plugin "parse_json" does not exist. Available options include [tags_on_failure, handle_failed_events, parse_when, source, pointer, destination, delete_source, overwrite_if_destination_exists]. Did you mean "destination"?
```

vs. 

```
1. entry-pipeline:processor:parse_json: caused by: Parameter "destintio" for plugin "parse_json" does not exist. Did you mean "destination"? Other available options include [tags_on_failure, handle_failed_events, parse_when, source, pointer, destination, delete_source, overwrite_if_destination_exists]. 
```

Reply:
Yea, that makes sense.

We probably need to make better use of newlines. Then it would be much cleaner.

```
entry-pipeline:processor:parse_json: caused by: 

Parameter "destintio" for plugin "parse_json" does not exist. Available options include [tags_on_failure, handle_failed_events, parse_when, source, pointer, destination, delete_source, overwrite_if_destination_exists]. 

Did you mean "destination"?
```